go-controller/pkg/ovn/subnetallocator/allocator.go

```diff
@@ -55,35 +55,57 @@ func (sna *SubnetAllocator) MarkAllocatedNetwork(subnet *net.IPNet) error {
 	return fmt.Errorf("network %s does not belong to any known range", subnet.String())
 }
 
-func maybeAllocateOneNetwork(ranges []*subnetAllocatorRange, networks []*net.IPNet) ([]*net.IPNet, error) {
-	if len(ranges) == 0 {
-		return networks, nil
+// AllocateNetworks tries to allocate networks in all the ranges available
+func (sna *SubnetAllocator) AllocateNetworks() ([]*net.IPNet, error) {
+	var networks []*net.IPNet
+	var err error
+	ipv4network, err := sna.AllocateIPv4Network()
+	if err != nil {
+		return nil, err
+	}
+	if ipv4network != nil {
+		networks = append(networks, ipv4network)
+	}
+	ipv6network, err := sna.AllocateIPv6Network()
+	if err != nil {
+		return nil, err
+	}
+	if ipv6network != nil {
+		networks = append(networks, ipv6network)
+	}
+	return networks, nil
+}
+
+// AllocateIPv4Network tries to allocate an IPv4 network if there are ranges available
+func (sna *SubnetAllocator) AllocateIPv4Network() (*net.IPNet, error) {
+	sna.Lock()
+	defer sna.Unlock()
+	if len(sna.v4ranges) == 0 {
+		return nil, nil
 	}
-	for _, snr := range ranges {
+	for _, snr := range sna.v4ranges {
 		sn := snr.allocateNetwork()
 		if sn != nil {
-			networks = append(networks, sn)
-			return networks, nil
+			return sn, nil
 		}
 	}
 	return nil, ErrSubnetAllocatorFull
 }
 
-func (sna *SubnetAllocator) AllocateNetworks() ([]*net.IPNet, error) {
```

Comment:
Ideally every commit should be a working state of the code; so you shouldn't remove this function without fixing its caller too.

Reply:
yeah, the github workflow is a bit weird, so I´m trying to do split the commits in these large PR for simplifying the reviews ... and squash them later